src/Strategy/Cocoapods/PodfileLock.hs

```diff
@@ -209,58 +214,66 @@ buildGraph lockFilePath lockFile@PodLock{lockExternalSources} = do
           --  * Parent has git source
           --
           -- We replace this dependency with parent's git sourced dependency.
-          let (parentSpec, candidateSubSpec) = splitOnceOn "/" dependencyName
+          let (parentSpec, childSpec) = splitOnceOn "/" dependencyName
 
-          if Data.Text.null candidateSubSpec
+          if Data.Text.null childSpec
             then pure d
             else do
               revisedDep <- case Map.lookup parentSpec lockExternalSources of
-                Just (ExternalPodSpec podSpecPath) -> fromMaybe d <$> (readPodSubSpecSourceAt podSpecPath parentSpec)
-                Just (ExternalPath podPath) -> fromMaybe d <$> (readPodSubSpecSourceAt (toPodSpecPath podPath parentSpec) candidateSubSpec)
+                Just (ExternalPodSpec podSpecPath) ->
+                  (fromMaybe d <$> (readPodSubSpecSourceAt podSpecPath childSpec)) <||> pure d
+                Just (ExternalPath podPath) ->
+                  (fromMaybe d <$> (readPodSubSpecSourceAt (toPodSpecPath podPath parentSpec) childSpec))
+                    <||> pure d
                 _ -> pure d
 
               when (revisedDep /= d) $
                 logDebug . pretty $
-                  "Replaced " <> dependencyName <> " with " <> parentSpec
-                    <> ", since "
-                    <> dependencyName
-                    <> " is subspec of "
-                    <> parentSpec
+                  "Replaced " <> dependencyName <> " with " <> parentSpec <> ", since " <> dependencyName <> " is subspec of " <> parentSpec
 
               pure revisedDep
       _ -> pure d
 
     readPodSubSpecSourceAt ::
       ( Has Exec sig m
       , Has Diagnostics sig m
+      , Has Logger sig m
       , Has (State PodSpecCache) sig m
       ) =>
       Text ->
       Text ->
       m (Maybe Dependency)
-    readPodSubSpecSourceAt podSpecPath candidateSubSpec = context ("Resolving vendored subspec of podspec at " <> showT podSpecPath) $ do
-      podSpecCache :: PodSpecCache <- get
-      (PodSpecJSON ExternalGitSource{urlOf, tagOf, commitOf, branchOf} subSpecs) <-
-        case Map.lookup podSpecPath podSpecCache of
-          Just cached -> pure cached
+    readPodSubSpecSourceAt podSpecPath candidateSubSpec = context
+      ( "Trying to resolve (" <> candidateSubSpec <> "), It is potentially a vendored subspec of podspec at: " <> showT podSpecPath
+      )
+      $ do
+        podSpecCache :: PodSpecCache <- get
```

Comment:
Does this really need a type annotation since there is only one `State` in the signature?